lib/src/main/java/com/cloudbees/groovy/cps/DepthTrackingEnv.java

```diff
@@ -0,0 +1,12 @@
+package com.cloudbees.groovy.cps;
+
+/**
+ * @author Sam Van Oort
+ */
+public interface DepthTrackingEnv extends Env {
+
+    /** Limit on how deeply environments can recurse */
+    int MAX_LEGAL_DEPTH = 3000;
```

Comment:
BTW how does this compare to default stack size for a typical JVM and effective recursion depth in that case?

Reply:
Varies depending on environment and settings.  In common cases, could be as little as 1000, or as much as 8000. 

In practice really all we care about is that "sane" code will run and that we catch runaway recursion -- also serializing the program will generate a StackOverflowError at some point (probably south of 1000 runs).